modules/app_people.rb

```diff
@@ -0,0 +1,49 @@
+require './student'
+require './teacher'
+
+module People
+  def list_all_people(people)
+    puts 'No person record created yet' if people.empty?
+    people.map { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+  end
+
+  def create_student(age, name, people)
+    print 'Has parent permission to? [Y/N]: '
+    permission = gets.chomp
+    yes_answers = %w[y yes]
+    no_answers = %w[n no]
+    if yes_answers.include? permission.downcase
+      student = Student.new(age, name, true)
+    elsif no_answers.include? permission.downcase
+      student = Student.new(age, name, false)
+    else
+      puts '=> Invalid input for permission'
+      return
+    end
+    people << student
+    puts 'Person created successfully'
+  end
+
+  def create_teacher(age, name, people)
+    print 'Specialization: '
+    specialization = gets.chomp
+    teacher = Teacher.new(age, name, specialization)
+    people << teacher
+    puts 'Person created successfully'
+  end
```

Comment:
- [ ] in this method you are printing to the console, getting data from the user and saving the data as well, this is not in line with the single responsibility rules, please have another look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) for more details.

Reply:
Hi, @mikethreels.
Thank you for the feedback. Do you mean that we should create a separate method for printing one line of code #28 and two more methods for the other parts of this create_teacher() method?
From our perspective, the create_teacher method is doing one job here, which is to create a teacher object by performing some micro-tasks.

Reply:
Thanks a lot for the clarification @Meltrust.